src/hotspot/share/gc/g1/g1_globals.hpp

```diff
@@ -379,6 +379,16 @@
            number of free regions and the expected survival rates in each   \
            section of the heap.")                                           \
                                                                             \
+  product(uint, G1RemoveSelfForwardPtrsWorkerCost, 2, DIAGNOSTIC,           \
+          "The factor for per region work cost to remove self forwardee "   \
+          "for evecuation failure regions.")                                \
+          range(1, 16)                                                      \
+                                                                            \
+  product(uint, G1RemoveSelfForwardPtrsThreadLoadFactor, 16, DIAGNOSTIC,    \
```

Comment:
I can't comprehend `G1RemoveSelfForwardPtrsThreadLoadFactor`, even after seeing how it is used in the impl. I get its purpose though: calculating a good chunk-size, small enough to have sufficient parallelism within a region but also large enough to keep bookkeeping/sync cost low. This reminds me of `G1RemSetScanState::get_chunks_per_region`. I wonder if we can reuse that heuristic for now.